androidApp/src/main/java/com/pulselink/data/messaging/FirebaseMessagingService.kt

```
@@ -47,6 +51,35 @@ class PulseLinkFirebaseMessagingService : FirebaseMessagingService() {
+                            val status = snapshot.getString("premiumSubscriptionStatus")
+                            val isPremium = status == "SUBSCRIPTION_STATE_ACTIVE" ||
+                                          status == "SUBSCRIPTION_STATE_IN_GRACE_PERIOD"
+                            val remoteWebAccess = snapshot.getBoolean("remoteWebAccessEnabled") ?: false
+
+                            settingsRepository.setPremiumUnlocked(isPremium)
+                            settingsRepository.setRemoteWebAccessEnabled(remoteWebAccess)
+
+                            Log.i(TAG, "Refreshed settings from cloud via FCM. Premium=$isPremium, Web=$remoteWebAccess")
+                            smsSyncTrigger.triggerSync()
+                        }
+                    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)